model/task/db.go

```diff
@@ -802,6 +802,105 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
+type TaskBuildVariants struct {
+	Task          string   `bson:"_id"`
+	BuildVariants []string `bson:"build_variants"`
+}
+
+// FindUniqueBuildVariantNamesByTask returns TaskBuildVariants with a list of unique build variants
+func FindUniqueBuildVariantNamesByTask(projectId string, taskName string) (*TaskBuildVariants, error) {
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			ProjectKey:     projectId,
+			DisplayNameKey: taskName,
+			RequesterKey:   bson.M{"$in": evergreen.SystemVersionRequesterTypes}},
+		}}
+
+	group := bson.M{
+		"$group": bson.M{
+			"_id":                 taskName,
+			BuildVariantKey + "s": bson.M{"$addToSet": "$" + BuildVariantKey},
+		},
+	}
+	unwindAndSort := []bson.M{
+		{
+			"$unwind": "$build_variants",
```

Comment:
Is the unwind necessary so that we can sort? Rather than unwinding, sorting, and then regrouping, can we just output the unsorted list and then sort it?

Reply:
I did it this way since i assumed it was the preferred  pattern to have the database sort. But if not i can make this change.